lang/LangSource/SC_LanguageConfig.cpp

```diff
@@ -93,47 +103,53 @@ bool SC_LanguageConfig::removeIncludedDirectory(const Path& path) { return remov
 
 bool SC_LanguageConfig::removeExcludedDirectory(const Path& path) { return removePath(mExcludedDirectories, path); }
 
+static void processPathList(const char* nodeName, YAML::Node& doc,
+                            const std::function<void(const boost::filesystem::path&)>& func) {
+    const YAML::Node& items = doc[nodeName];
+    if (items && items.IsSequence()) {
+        for (auto const& item : items) {
+            const std::string& path = item.as<std::string>("");
+            if (!path.empty()) {
+                const boost::filesystem::path& native_path = SC_Codecvt::utf8_str_to_path(path);
+                func(native_path);
+            }
+        }
+    }
+}
+
+static void processBool(const char* nodeName, YAML::Node& doc, const std::function<void(bool)>& successFunc,
+                        const std::function<void(void)>& failFunc) {
+    const YAML::Node& item = doc[nodeName];
+    if (item) {
+        try {
+            successFunc(item.as<bool>());
+            return;
+        } catch (...) {
+            postfl("Warning: Cannot parse config file entry \"%s\"\n", nodeName);
+            failFunc();
+        }
+    }
+    failFunc(); // call failFunc if item is not found
```

Comment:
this should be placed in the `else` branch otherwise it will be called twice if line 125 throws. you can also remove the `return` in the `try` block then.

Reply:
just saw @elgiano 's comment above. you didn't quite use the diff he suggested, which was to delete line 129 entirely. you could also do that. overall, i think the control flow is clearer with my suggestion, but it's up to you.